lighthouse-core/lib/asset-saver.js

```diff
@@ -119,6 +120,10 @@ function prepareAssets(artifacts, results) {
         const traceData = Object.assign({}, trace);
         const html = screenshotDump(screenshots, results);
 
+        if (results && results.audits) {
+          const evts = new Metrics(traceData.traceEvents, results.audits).generateFakeEvents();
+          traceData.traceEvents.push(...evts);
```

Comment:
It seems bad to alter the original trace. Not sure when we did `const traceData = Object.assign({}, trace);`, but really we should be doing something like `traceData.traceEvents = Array.from(traceData.traceEvents)` to get a new array. Is that crazy?